src/rewards/cross-chain/RewardsMerkle.sol

```diff
@@ -42,29 +42,36 @@ abstract contract RewardsMerkle is RewardsStorage, MessageSenderApp {
     event AllSubRootsGenerated(uint64 indexed epoch);
     event TopRootGenerated(uint64 indexed epoch, bytes32 topRoot);
     event TopRootSent(uint64 indexed epoch, bytes32 topRoot, address receiver, uint64 dstChainId);
+    event GlobalStateUpdated(uint64 indexed currEpoch, State indexed state);
     event MessageBusUpdated(address messageBus);
 
     // ----------- state transition -----------
+
     function startEpoch(uint64 epoch) external onlyRole(REWARD_UPDATER_ROLE) {
         require(state == State.Idle, "invalid state");
         require(epoch > currEpoch, "invalid epoch");
         currEpoch = epoch;
         state = State.RewardsSubmission;
-        subRoots.clear();
+        emit EpochStarted(epoch);
+    }
+
+    function restartEpoch(uint64 epoch) external onlyRole(REWARD_UPDATER_ROLE) {
+        require(state == State.Idle, "invalid state");
+        require(epoch == currEpoch, "can only restart current epoch");
```

Comment:
Maybe change to `require(epoch >= currEpoch, "invalid epoch");` in `startEpoch`, and remove this function?